### PR TITLE
Use LogicalBinOp to compute the result of CaseStmt

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/libasr/pass/pass_utils.cpp [hunk 1]
```diff
@@ -458,17 +458,18 @@ namespace LCompilers {
         ASR::expr_t* create_compare_helper(Allocator &al, const Location &loc, ASR::expr_t* left, ASR::expr_t* right,
                                                 ASR::cmpopType op) {
             ASR::ttype_t* type = ASRUtils::expr_type(left);
+            ASR::ttype_t* cmp_type = ASRUtils::TYPE(ASR::make_Logical_t(al, loc, 4, nullptr, 0));
             // TODO: compute `value`:
             if (ASRUtils::is_integer(*type)) {
-                return ASRUtils::EXPR(ASR::make_IntegerCompare_t(al, loc, left, op, right, type, nullptr));
+                return ASRUtils::EXPR(ASR::make_IntegerCompare_t(al, loc, left, op, right, cmp_type, nullptr));
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/pass_utils.cpp:464]: Any `Compare` node must return `logical type`, right?
@certik @czgdp1807

**Comment by Thirumalai-Shaktivel** [comment]
This generates the following to address https://github.com/lfortran/lfortran/pull/1326#discussion_r1114406148
```
(LogicalBinOp
    (StringCompare
        (StringConstant
            "a"
            (Character 1 1 () [])
        )
        LtE
        (Var 2 c)
        (Logical 4 [])
        ()
    )
    And
    (StringCompare
        (Var 2 c)
        LtE
        (StringConstant
            "j"
            (Character 1 1 () [])
        )
        (Logical 4 [])
        ()
    )
    (Logical 4 [])
    ()
)
```

File: src/libasr/pass/pass_utils.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/pass/pass_utils.cpp:464]: I think so.

**Review by certik — APPROVED** [review]
Looks good to me.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review.

**Pull request merged**